titus-server-master/src/main/java/com/netflix/titus/master/jobmanager/service/batch/BatchDifferenceResolver.java

```diff
@@ -269,17 +269,19 @@ public BatchDifferenceResolver(
         for (BatchJobTask refTask : tasks) {
             BatchJobTask runningTask = runningJobView.getTaskById(refTask.getId());
             if (JobFunctions.isOwnedByKubeScheduler(refTask)) {
-                // TODO This complexity exists due to the way Fenzo is initialized on bootstrap. This code can be simplified one we move off Fenzo.
-                if (runningTask == null || (refTask.getStatus().getState() == TaskState.Accepted && !TaskStatus.hasPod(refTask))) {
-                    missingTasks.add(BasicTaskActions.launchTaskInKube(
-                            kubeApiServerIntegrator,
-                            configuration,
-                            engine,
-                            runningJobView.getJob(),
-                            refTask,
-                            RECONCILER_CALLMETADATA.toBuilder().withCallReason("Launching task in Kube").build(),
-                            titusRuntime
-                    ));
+                if (kubeApiServerIntegrator.isReadyForScheduling()) {
```

Comment:
minor: combine the `if` below into a single boolean logic operation (`&&`)

Reply:
even better, maybe consider extracting the boolean logic chain into a separate method/function

Reply:
I was thinking about extracting this into a util function, but there would be too many arguments, which would not improved the code readability.